fdbserver/workloads/ClogRemoteTLog.actor.cpp

```diff
@@ -175,14 +176,52 @@ struct ClogRemoteTLog : TestWorkload {
 		}
 	}
 
-	ACTOR static Future<bool> grayFailureStatusCheck(Database db, NetworkAddress cloggedRemoteTLog) {
-		StatusObject status = wait(StatusClient::statusFetcher(db));
-		StatusObjectReader reader(status);
+	// Returns true if and only if there's a general error in fetching status json
+	// An example failure is network issue from client to CC (server)
+	static bool statusError(StatusObjectReader reader) {
+		static const auto errors{ []() {
+			std::unordered_set<std::string> errors;
+			std::for_each(messageTypeToName.begin(), messageTypeToName.end(), [&errors](const auto& kvPair) {
+				errors.insert(kvPair.second);
+			});
+			return errors;
+		}() };
+
+		StatusObjectReader client;
+		if (!reader.get("client", client)) {
+			TraceEvent("NoClient");
+			return true;
+		}
+
 		StatusObjectReader cluster;
 		if (!reader.get("cluster", cluster)) {
 			TraceEvent("NoCluster");
-			return false;
+			return true;
+		}
+
+		ASSERT(client.has("messages"));
+		StatusArray messages = client["messages"].get_array();
+		for (StatusObjectReader message : messages) {
+			if (message.has("name") && errors.contains(message["name"].get_str())) {
+				return true;
+			}
 		}
+
+		return false;
+	}
+
+	ACTOR static Future<bool> grayFailureStatusCheck(Database db, NetworkAddress cloggedRemoteTLog) {
+		StatusObject status = wait(StatusClient::statusFetcher(db));
+		StatusObjectReader reader(status);
+
+		if (statusError(reader)) {
+			// If there is some error to get the status (e.g. network issue), we let gray failure status check pass
```

Comment:
Should we adda TraceEvent here for better observability of those tests?

Reply:
Ah, github mobile doesn't show me comments upfront so I missed this. Since the test will pass if there is a general error, not sure how useful the trace event would be, but doesn't hurt to add either. Will quickly send you a follow up PR.

Reply:
https://github.com/apple/foundationdb/pull/11803 addresses this. I noticed `statusError` implementation already had trace events for all cases except one place, which is where I added the trace event. That way, we get richer information about what failed which is not available at line 218 here.